klippa/src/lib.rs

```diff
@@ -508,6 +501,16 @@ pub trait NameIdClosure {
     fn collect_name_ids(&self, plan: &mut Plan);
 }
 
+pub trait Subset {
+    /// Subset this table, if any error occurs, the table will be dropped
```

Comment:
Dropped? Or just not added to builder in the first place?

Optional alternate phrasing:

`/// if successful a subset version of this table will have been added to builder`